sre_yield/__init__.py

```diff
@@ -216,16 +216,17 @@ def get_item(self, i, d=None):
         if i < 0 or i >= self.length:
             raise IndexError("Index %d out of bounds" % (i,))
 
-        if len(self.list_lengths) == 1:
-            # skip unnecessary ''.join -- big speedup
-            return self.list_lengths[0][0][i]
-
         for c, c_len in self.list_lengths:
             i, mod = divmod(i, c_len)
             if hasattr(c, "get_item"):
                 result.append(c.get_item(mod, d))
             else:
                 result.append(c[mod])
+
+        if len(result) == 1:
+            # skip unnecessary ''.join -- big speedup
```

Comment:
I wonder if there is much speed up after this bug fix.
If the perf hit was mostly in "".join for a single item, my guess is that it has been fixed in Python 3.

I see you do have some bench tools already hooked up, so I am keen to play with them to investigate this, but I've got a few more patches to tidy up before I come back to these.

Reply:
I agree, in this position it should not be a measurable difference.  In its previous position, it avoided a divmod on arbitrary-precision integers.  You can remove this optimization.